src/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -151,7 +156,8 @@ See the LICENSE file in the project root for more information.
       <IlcArg Condition="$(DebugSymbols) == 'true'" Include="-g" />
       <IlcArg Condition="$(IlcGenerateMapFile) == 'true'" Include="--map:$(NativeIntermediateOutputPath)%(ManagedBinary.Filename).map.xml" />
       <IlcArg Condition="$(RdXmlFile) != ''" Include="--rdxml:$(RdXmlFile)" />
-      <IlcArg Condition="$(OutputType) ==  'Library' and $(NativeLib) != ''" Include="--nativelib" />
+      <IlcArg Condition="$(OutputType) == 'Library' and $(NativeLib) != ''" Include="--nativelib" />
+      <IlcArg Condition="$(OutputType) == 'Library' and $(NativeLib) == 'Shared'" Include="--exportsfile:$(ExportsFile)" />
```

Comment:
I would change the condition to `$(ExportsFile) != ''` to make it simpler and more flexible